### PR TITLE
Migrate examples to use pallet macro

### PR DESCRIPTION
**Pull request by thiolliere**
part of https://github.com/paritytech/substrate/issues/7882

migrate pallet-example, pallet-example-offchain-worker and pallet-example-parallel to pallet macro.

the doc in pallet-example is slightly changed. Also it might be good to also have inherent, validate unsigned, and error docs but this is for later PR.

At some point I think the doc in frame_support::pallet and pallet-example should be merged

### CODE REVIEW

**Review by ascjones — APPROVED** [review]
LGTM

**Review by kianenigma — APPROVED** [review]
LGTM

File: frame/balances/src/weights.rs [hunk 1]
```diff
@@ -18,7 +18,7 @@
 //! Autogenerated weights for pallet_balances
 //!
 //! THIS FILE WAS AUTO-GENERATED USING THE SUBSTRATE BENCHMARK CLI VERSION 2.0.0
-//! DATE: 2021-01-06, STEPS: [50, ], REPEAT: 20, LOW RANGE: [], HIGH RANGE: []
+//! DATE: 2021-01-06, STEPS: \[50, \], REPEAT: 20, LOW RANGE: [], HIGH RANGE: []
```
Comment (shawntabrizi) [inline frame/balances/src/weights.rs:21]: where is this change coming from?

do you need to update the weight template?

Comment (thiolliere) [reply frame/balances/src/weights.rs:21]: yes we probably need to update the templates,
rust doc complains the content inside `[]` is not a rust type

**Comment by thiolliere** [comment]
bot merge

**Pull request merged**